specs/interop.md

```diff
@@ -0,0 +1,396 @@
+of the initiating transaction when verifying the relaying transaction.
+
+#### Initiating Messages
+
+An initiating message emits the following event:
+
+```solidity
+event MessagePassed(
+    bytes32 indexed messageHash,
+    address indexed sender,
+    address indexed target,
+    uint256 nonce,
+    uint256 value,
+    uint256 timestamp,
+    uint256 chainid,
+    bytes data
+);
+```
+
+The chain id is of the source chain. Note that the chain id of the destination chain is not specified
+as part of the cross chain message. This means that the cross chain message is valid to be played
+on any destination chain. Binding to a particular destination chain can be implemented at the application layer.
+
+The `messageHash` and chain id uniquely identify the existence of an initiating message and SHOULD
+be used by a block builder to ensure the existence of the initiating message when verifying the relaying
+transaction. `messageHash` SHOULD be `indexed` to allow for easy filtering of logs.
+
+A cross chain message is defined in the following format:
+
+```solidity
+struct CrossChainMessage {
+    uint256 nonce;
+    address sender;
+    address target;
+    uint256 value;
+    uint256 timestamp;
+    uint256 chainid;
+    bytes data;
+}
+```
+
+Note that there is no `gasLimit` to avoid complexity with EVM gas introspection.
+
+The `messageHash` is constructed like so:
+
+```solidity
+function hashCrossChainMessage(CrossChainMessage memory _msg) pure returns (bytes32) {
+    return keccak256(abi.encode(_msg.nonce, _msg.sender, _msg.target, _msg.value, _msg.timestamp, _msg.chainid, _msg.data));
+}
+```
+
+### CrossL2Inbox
+
+Address: `0x4200000000000000000000000000000000000022`
+
+The `CrossL2Inbox` is responsible for relaying a cross chain message on the destination chain.
+It is permissionless to finalize a cross chain message on behalf of any user. Certain protocol
+enforced invariants must be preserved to ensure safety of the protocol.
+
+#### Invariants
+
+- The timestamp of relaying message MUST be greater than or equal to the timestamp of the initiating message
+- The chain id of the initiating message MUST be in the dependency set
+- The relaying message MUST be initiated by an externally owned account such that the top level EVM call frame enters the `CrossL2Inbox`
+
+##### Only EOA Invariant
+
+The `onlyEOA` invariant on relaying a cross chain message enables static analysis on relaying messages.
+This allows for the derivation pipeline and block builders to reject relaying messages that do not
+have a corresponding initiating message.
+
+##### Transferring Ether in a Cross Chain Message
+
+The `CrossL2Inbox` MUST be initially set in state with an ether balance of `type(uint256).max`.
+There is no reason for a user to send [value](https://github.com/ethereum/execution-specs/blob/1fed0c0074f9d6aab3861057e1924411948dc50b/src/ethereum/frontier/fork_types.py#L53)
```

Comment:
This isn't a particularly reassuring way to phrase this.  Might be worth noting that transactions that cause a balance overflow revert. Reverting those transaction is ok because there's no reason for a user to send value...

Reply:
The spec has changed and this no longer applies